WebRoot/js/zimbraMail/share/view/dialog/ZmUploadDialog.js

```diff
@@ -479,28 +477,37 @@ ZmUploadDialog.prototype._addFileInputRow = function(oneInputOnly) {
     }
 };
 
-ZmUploadDialog.prototype._handleFileSize =
-function(inputEl, sizeEl){
+ZmUploadDialog.prototype._handleFileSelection =
+function(inputEl, fileDetailEl){
 
     var files = inputEl.files;
-    if(!files) return;
+    if(!files || !files.length){
+        fileDetailEl && (fileDetailEl.innerHTML = "");
+        return;
+    }
 
-    var sizeStr = [], className, totalSize =0;
+    var sizeStr = [], className, fileNameDetail, totalSize =0;
     for(var i=0; i<files.length;i++){
         var file = files[i];
         var size = file.size || file.fileSize /*Safari*/ || 0;
-	    var aCtxt = ZmAppCtxt.handleWindowOpener();
+        var aCtxt = ZmAppCtxt.handleWindowOpener();
         if(size > aCtxt.get(ZmSetting.DOCUMENT_SIZE_LIMIT))
             className = "RedC";
         totalSize += size;
     }
 
-    if(sizeEl) {
-        sizeEl.innerHTML = "  ("+AjxUtil.formatSize(totalSize, true)+")";
+    if(files.length > 1) {
```

Comment:
What I understand from the link is it can lead to performance improvement with caching the required object/array, in cases where we are using nested objects or objects having arrays.

In my case I don't have a nested structure anywhere, I am not convinced that caching a **property**  present on the object into a new variable would serve any purpose.!!

As far as debugging is concerned, hovering over the arrayWhat I understand from the link is it can lead to performance improvement with caching the required object/array, in cases where we are using nested objects or objects having arrays.

In my case I don't have a nested structure anywhere, I am not convinced that caching a **property**  present on the object into a new variable would serve any purpose.!!

As far as debugging is concerned, hovering over the array/object shows up all the properties/elements.
There's no rule defined that says Cache everything!!!